File: multilibrary/src/main/java/com/zhongjh/multimedia/BaseFragment.java
package com.zhongjh.multimedia;

import android.view.KeyEvent;

import androidx.fragment.app.Fragment;

import com.zhongjh.multimedia.listener.HandleFragmentInterface;
import com.zhongjh.multimedia.utils.HandleBackUtil;
import com.zhongjh.multimedia.listener.HandleFragmentInterface;
import com.zhongjh.multimedia.utils.HandleBackUtil;

import org.jetbrains.annotations.NotNull;

/**
 * 录音、视频、音频的fragment继承于他
 * @author zhongjh
 */
public abstract class BaseFragment extends Fragment implements HandleFragmentInterface {

    @Override
    public boolean onBackPressed() {
        return HandleBackUtil.handleBackPress(this);
    }

    @Override
    public boolean onKeyDown(int keyCode, @NotNull KeyEvent event) {
        return HandleBackUtil.handleBackPress(this);
    }


}

File: multilibrary/src/main/java/com/zhongjh/multimedia/BaseFragment.kt
package com.zhongjh.multimedia

import android.view.KeyEvent
import androidx.fragment.app.Fragment
import com.zhongjh.multimedia.listener.HandleFragmentInterface
import com.zhongjh.multimedia.utils.HandleBackUtil.handleBackPress

/**
 * 录音、视频、音频的fragment继承于他
 * @author zhongjh
 */
abstract class BaseFragment : Fragment(), HandleFragmentInterface {
    override fun onBackPressed(): Boolean {
        return handleBackPress(this)
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return handleBackPress(this)
    }
}
